darshan-util/pydarshan/benchmarks/benchmarks/cffi_backend.py

```diff
@@ -0,0 +1,33 @@
+from darshan.backend import cffi_backend
+from darshan.log_utils import get_log_path
+
+
+class RetrieveLogData:
+    # see gh-811
+
+    # TODO: maybe produce a version of the log that also
+    # has runtime HEATMAP to guard memory leaks on that module;
+    # part of the issue here is
+    # perhaps the following to pull in new log repo files:
+    # https://github.com/darshan-hpc/darshan-logs/issues/15
+
+    params = ['POSIX', 'MPI-IO', 'H5F', 'H5D', 'LUSTRE', 'STDIO', 'DXT_POSIX', 'DXT_MPIIO']
+    param_names = ["module"]
+
+
+    def setup(self, module):
+        self.ior_log_path = get_log_path("ior_hdf5_example.darshan")
+
+
+    def peakmem_log_get_record(self, module):
+        for i in range(100000):
+            log = cffi_backend.log_open(self.ior_log_path)
+            rec = cffi_backend.log_get_record(log, module)
+            cffi_backend.log_close(log)
+
+
+    def peakmem_log_get_mounts(self, module):
+        for i in range(100000):
```

Comment:
The magnitude of the mount leak is so massive we could probably cut this down by 10x so we don't hit peak memory values above the capacity of an average laptop. The current ratios are `0.00` since they span orders of magnitude improvements.